lib/matplotlib/axis.py

```diff
@@ -2619,7 +2625,8 @@ def set_ticks_position(self, position):
             self.set_tick_params(which='both', right=True, labelright=False,
                                  left=True, labelleft=True)
         else:
-            assert False, "unhandled parameter not caught by _check_in_list"
+            _api.check_in_list(['left', 'right', 'both', 'default', 'none'],
+                               position=position)
```

Comment:
This halves the number of calls to `check_in_list`, from ~100.000 to ~50.000 in the example. For cases like this, where it is called from a commonly used `__init__`-method  it is probably better to have it last. Although, in general error checking may be better off early.